scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -269,19 +359,21 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
         if (lastNewlinePos != -1 &&
           prev != null && prev.is[CanEndStat] &&
           next != null && next.isNot[CantStartStat] &&
-          (sepRegions.isEmpty || sepRegions.head == '}' || sepRegions.head == '$')) {
+          (next.text != "||" && next.text != "&&" && next.text != ":::") &&
```

Comment:
Why do we need to add these? And not for example `::`

Reply:
changed to `isOperatorPart` as discussed :+1: